### DISCUSSION TITLE
Trying to implement OpenZeppelin UUPS in an NFT On-Chain Project

### DISCUSSION THREAD

**Discussion by yeahChibyke** [Cyfrin/foundry-full-course-cu#2504] [Q&A]
I am building an upgradeable fully on-chain NFT project. All is going well so far, but I am unable to effectively call the initialize function of the upgrade contract. I am testing the project, and anytime I want to test that the upgrade is effective, it fails.

Here is a gits of the contracts, scripts, and tests:
https://gist.github.com/yeahChibyke/a1155fe37272e075d86dca4a5cdad419

And here is the repo:
https://github.com/yeahChibyke/Caterpillar-Butterfly

**Comment by EngrPips** [comment]
Hello @yeahChibyke, I must say this discussion channel is primarily to help people who run into issues while taking the content of the course associated with this GitHub repo, but still, I hope someone who knows enough and is willing to help see this and help you resolve this issue.

**Reply by yeahChibyke** [reply]
@EngrPips I am using the Updraft course to learn, and I always try to build a similar project with whatever I am learning at the moment, but with a certain level of difficulty so as to challenge myself. I just finished the Upgradeable Contracts 2 days ago, and I immediately decided to build an upgradeable on-chain NFT contract yesterday. I have searched all over GitHub to see if I can find a similar project I can use as resource. No luck. Hence I brought it here. 🙏🏾

**Reply by EngrPips** [reply]
You can give more specific details as to what part of your code is failing and send the test output so that we can take a glance and see how much we can help rather than sending a whole repo. I mean, many people might not have that much time to start searching through a whole repo.

**Reply by yeahChibyke** [reply]
Here is a gist that has the essential aspects of the project (nft contracts, deploy scripts, tests):
https://gist.github.com/yeahChibyke/a1155fe37272e075d86dca4a5cdad419

I posted the full repo in the original  post so that whoever wanted to replicate the project will be aware of the dependencies to install

**Reply by EngrPips** [reply]
Alright, Thanks for the heads-up.
